2011-05-06  Richard Sandiford  <richard.sandiford@linaro.org>

	gcc/
	From Sergey Grechanik  <mouseentity@ispras.ru>, approved for mainline

	* config/arm/arm.c (coproc_secondary_reload_class): Return NO_REGS
	for constant vectors.

=== modified file 'gcc/config/arm/arm.c'
--- old/gcc/config/arm/arm.c	2011-05-03 15:18:07 +0000
+++ new/gcc/config/arm/arm.c	2011-05-06 11:33:02 +0000
@@ -9193,7 +9193,7 @@
   /* The neon move patterns handle all legitimate vector and struct
      addresses.  */
   if (TARGET_NEON
-      && MEM_P (x)
+      && (MEM_P (x) || GET_CODE (x) == CONST_VECTOR)
       && (GET_MODE_CLASS (mode) == MODE_VECTOR_INT
 	  || GET_MODE_CLASS (mode) == MODE_VECTOR_FLOAT
 	  || VALID_NEON_STRUCT_MODE (mode)))

